python/hail/expr/expression.py

```diff
@@ -529,6 +530,157 @@ def __ne__(self, other):
+    def show(self, n=10, width=90, truncate=None, types=True):
+        """Print the first few rows of the table to the console.
+
+        Examples
+        --------
+        .. doctest::
+
+            >>> table1.SEX.show()
+            +-------+--------+
+            |    ID | SEX    |
+            +-------+--------+
+            | Int32 | String |
+            +-------+--------+
+            |     1 | M      |
+            |     2 | M      |
+            |     3 | F      |
+            |     4 | F      |
+            +-------+--------+
+
+            >>> hl.capture(123).show()
+            +--------+
+            | <expr> |
+            +--------+
+            |  Int32 |
+            +--------+
+            |    123 |
+            +--------+
+
+        Warning
+        -------
+        Extremely experimental.
+
+        Parameters
+        ----------
+        n : :obj:`int`
+            Maximum number of rows to show.
+        width : :obj:`int`
+            Horizontal width at which to break columns.
+        truncate : :obj:`int`, optional
+            Truncate each field to the given number of characters. If
+            ``None``, truncate fields to the given `width`.
+        types : :obj:`bool`
+            Print an extra header line with the type of each field.
+        """
+        name = None
+        source = self._indices.source
+        if source is not None:
+            fields_rev = {v: k for k, v in source._fields.items()}
+            name = fields_rev.get(self, None)
+        t = self._to_table()
+        if name is not None:
+            if name in t.key:
+                t = t.select(name)
+            else:
+                t = t.rename({'<expr>': name})
+        t.show(n, width, truncate, types)
+
+    @handle_py4j
+    @typecheck_method(n=int)
+    def take(self, n):
+        """Collect the first `n` records of an expression.
+
+        Examples
+        --------
+        Take the first three rows:
+
+        .. doctest::
+
+            >>> first3 = table1.X.take(3)
+            [5, 6, 7]
+
+        Warning
+        -------
+        Extremely experimental.
+
+        Parameters
+        ----------
+        n : int
+            Number of records to take.
+
+        Returns
+        -------
+        :obj:`list`
+        """
+        return [r['<expr>'] for r in self._to_table().take(n)]
+
+    @handle_py4j
+    def collect(self):
+        """Collect all records of an expression into a local list.
+
+        Examples
+        --------
+        Take the first three rows:
+
+        .. doctest::
+
+            >>> first3 = table1.C1.collect()
+            [2, 2, 10, 11]
+
+        Warning
+        -------
+        Extremely experimental.
+
+        Warning
+        -------
+        The list of records may be very large.
+
+        Returns
+        -------
+        :obj:`list`
+        """
+        return [r['<expr>'] for r in self._to_table().collect()]
```

Comment:
I feel like maybe _to_table should take the field name, so you can do something a little more readable like `[r.x for r in self._to_table('x').collect()]`?